### PR TITLE
Remove WorkHistoryBreak#application_form_id

### PR DESCRIPTION
**Pull request by CatalinVoineag**
## Context

Currently the application_work_history_break is linked to an application_form through the breakable polymorphic column

This commit removes the old link, through the application_form_id, the column will be dropped in a future PR.

We also need to touch the application_form and application_choices like the old relation did, to replicate the old behaviour

The backfill_application_work_history_breaks
data migration needs to be removed as well, no longer needed

## Changes proposed in this pull request

Remove application_form_id link between application_work_history_breaks and application_form

## Guidance to review

Create/edit a work history break on local or review

https://github.com/user-attachments/assets/54d79f80-9c30-413b-8f35-1a4de6246585

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)
- [x] Inform data insights team due to database changes
- [x] Make sure all information from the Trello card is in here
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Add PR link to Trello card

### CODE REVIEW

**Comment by CatalinVoineag** [comment]
[Edit work experiences - Change db structure to accommodate work experience edits](https://trello.com/c/VzPy9EWy/2163-edit-work-experiences-change-db-structure-to-accommodate-work-experience-edits)

File: app/models/application_work_history_break.rb [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
-  belongs_to :breakable, polymorphic: true
+  belongs_to :breakable, polymorphic: true, touch: true
 
   before_save -> { self.application_form_id = breakable_id }, if: -> { application_form_id.nil? }
 
-  audited associated_with: :application_form
+  audited associated_with: :breakable
+
+  after_commit do
+    if application_form
```
Comment (avinhurry) [inline app/models/application_work_history_break.rb:9]: How about `if application_form_breakable`?

**Review by avinhurry — APPROVED** [review]

File: app/models/application_work_history_break.rb (hunk 1 above)
Comment (CatalinVoineag) [reply app/models/application_work_history_break.rb:9]: I think `application_form` is a better method name as it's a public method that can be used by other things, `application_work_history_break.application_form` you just want the application form, you don't really care how it's linked to it

Comment (avinhurry) [reply app/models/application_work_history_break.rb:9]: Sounds good 👍🏼

**Pull request merged**